apm-agent-core/src/test/java/co/elastic/apm/agent/MockReporter.java

```diff
@@ -80,13 +80,13 @@ private static JsonSchema getSchema(String resource) {
     }
 
     @Override
-    public void report(Transaction transaction) {
+    public synchronized void report(Transaction transaction) {
```

Comment:
what's the benefit of synching here? isn't there still a race condition?

Reply:
Does this change the visibility status? The Transactions/Spans/Errors go into containers, for which the reference doesn't change. I would assume making these collections synchronize be more suitable, but if it works- it works 😄

Reply:
Right, making the collections synchronized might be better